specification/cognitiveservices/OpenAI.Inference/models/completions/chat_messages.tsp

```diff
@@ -0,0 +1,202 @@
+  @doc("The chat role associated with this message, which is always 'assistant' for assistant messages.")
+  role: ChatRole.assistant;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "explicitly nullable in mirrored API"
+  @doc("The content of the message.")
+  content: string | null;
+
+  @doc("An optional name for the participant.")
+  name?: string;
+
+  @projectedName("json", "tool_calls")
+  @added(ServiceApiVersions.v2023_12_01_Preview)
+  @doc("""
+  The tool calls that must be resolved and have their outputs appended to subsequent input messages for the chat
+  completions request to resolve as configured.
+  """)
+  toolCalls?: ChatCompletionsToolCall[];
+
+  @added(ServiceApiVersions.v2023_07_01_Preview)
+  @removed(ServiceApiVersions.v2023_12_01_Preview)
+  @projectedName("json", "function_call")
+  @doc("""
+  The function call that must be resolved and have its output appended to subsequent input messages for the chat
+  completions request to resolve as configured.
+  """)
+  functionCall?: FunctionCall;
+}
+
+@added(ServiceApiVersions.v2023_12_01_Preview)
+@doc("A request chat message representing requested output from a configured tool.")
+model ChatRequestToolMessage extends ChatRequestMessage {
+  @doc("The chat role associated with this message, which is always 'tool' for tool messages.")
+  role: ChatRole.tool;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "explicitly nullable in mirrored API"
+  @doc("The content of the message.")
+  content: string | null;
+
+  @projectedName("json", "tool_call_id")
+  @doc("The ID of the tool call resolved by the provided content.")
+  toolCallId: string;
+}
+
+@added(ServiceApiVersions.v2023_07_01_Preview)
+@removed(ServiceApiVersions.v2023_12_01_Preview)
+@doc("A request chat message representing requested output from a configured function.")
+model ChatRequestFunctionMessage extends ChatRequestMessage {
+  @doc("The chat role associated with this message, which is always 'function' for function messages.")
+  role: ChatRole.function;
+
+  @doc("The name of the function that was called to produce output.")
+  name: string;
+
+  @doc("The output of the function as requested by the function call.")
+  content: string;
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("A representation of a chat message as received in a response.")
+model ChatResponseMessage {
+  @doc("The chat role associated with the message.")
+  role: ChatRole;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "explicitly nullable in mirrored API"
+  @doc("The content of the message.")
+  content: string | null;
+
+  @added(ServiceApiVersions.v2023_12_01_Preview)
+  @projectedName("json", "tool_calls")
+  @doc("""
+  The tool calls that must be resolved and have their outputs appended to subsequent input messages for the chat
+  completions request to resolve as configured.
+  """)
+  toolCalls?: ChatCompletionsToolCall[];
```

Comment:
I think that the yaml uses specifically the concretization `ChatCompletionsFunctionToolCall`. Seems to always be a `chatCompletionMessageToolCall` (yaml file name)